Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ClientReconnectingState.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Collections;
+using System.Threading.Tasks;
+using Unity.Multiplayer.Samples.BossRoom.Shared;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Net.UnityServices.Lobbies;
+using Unity.Multiplayer.Samples.Utilities;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to a client attempting to reconnect to a server. It will try to reconnect a
+    /// number of times defined by k_NbReconnectAttempts. If it succeeds, it will transition to the Connected state. If
+    /// not, it will transition to the Offline state.
+    /// </summary>
+    public class ClientReconnectingState : OfflineState
+    {
+        const int k_NbReconnectAttempts = 2;
+
+        IPublisher<ReconnectMessage> m_ReconnectMessagePublisher;
+
+        Coroutine m_ReconnectCoroutine;
+        string m_LobbyCode = "";
+        int m_NbAttempts;
+
+        [Inject]
+        void InjectDependencies(ProfileManager profileManager, LobbyServiceFacade lobbyServiceFacade, LocalLobby localLobby, IPublisher<ReconnectMessage> reconnectMessagePublisher, IPublisher<ConnectStatus> connectStatusPublisher)
+        {
+            m_ReconnectMessagePublisher = reconnectMessagePublisher;
+            base.InjectDependencies(profileManager, lobbyServiceFacade, localLobby, connectStatusPublisher);
+        }
+
+        public override void Enter()
+        {
+            m_LobbyCode = m_LobbyServiceFacade.CurrentUnityLobby != null ? m_LobbyServiceFacade.CurrentUnityLobby.LobbyCode : "";
+            m_ReconnectCoroutine = m_ConnectionManager.StartCoroutine(ReconnectCoroutine());
+            m_NbAttempts = 0;
+        }
+
+        public override void Exit()
+        {
+            if (m_ReconnectCoroutine != null)
+            {
+                m_ConnectionManager.StopCoroutine(m_ReconnectCoroutine);
+                m_ReconnectCoroutine = null;
+            }
+            m_ReconnectMessagePublisher.Publish(new ReconnectMessage(k_NbReconnectAttempts, k_NbReconnectAttempts));
+        }
+
+        public override void OnClientConnected(ulong clientId)
```

Comment:
I clientID as a parameter here, I'd assume this gets triggered when other clients connect as well. Did you test a case with more than a one client?

Reply:
The docs mention that "this callback is only ran on the server and on the local client that connects." But even if it was also ran on other clients, if we are here, we are not connected to the server anyways, so this couldn't be triggered when other clients connect to the server